tests/Versioning/VersionManagerTest.php

```diff
@@ -420,6 +449,54 @@ public function testRestoreWithNewChildren()
         $this->assertFalse($node->hasNode('newChildNode'));
     }
 
+    public function testRestoreWithChildrenInitialize()
+    {
+        $nodeTypeManager = $this->session->getWorkspace()->getNodeTypeManager();
+
+        $wildcardPropertyDefinition = $nodeTypeManager->createPropertyDefinitionTemplate();
+        $wildcardPropertyDefinition->setName('*');
+
+        $testNodeType = $nodeTypeManager->createNodeTypeTemplate();
+        $testNodeType->setName('nt:test-initialize-children');
+        $testNodeType->getPropertyDefinitionTemplates()->append($wildcardPropertyDefinition);
+
+        $testChildNodeType = $nodeTypeManager->createNodeTypeTemplate();
+        $testChildNodeType->setName('nt:child');
+        $testChildNodeType->getPropertyDefinitionTemplates()->append($wildcardPropertyDefinition);
+
+        $childNodeDefinition = $nodeTypeManager->createNodeDefinitionTemplate();
+        $childNodeDefinition->setName('child');
+        $childNodeDefinition->setDefaultPrimaryTypeName('nt:child');
+        $childNodeDefinition->setRequiredPrimaryTypeNames(['nt:child']);
+        $childNodeDefinition->setAutoCreated(true);
+        $childNodeDefinition->setMandatory(true);
+        $childNodeDefinition->setOnParentVersion(OnParentVersionAction::INITIALIZE);
+        $testNodeType->getNodeDefinitionTemplates()->append($childNodeDefinition);
+
+        $nodeTypeManager->registerNodeType($testChildNodeType, true);
+        $nodeTypeManager->registerNodeType($testNodeType, true);
```

Comment:
are the other tests working? if not, it might be a problem with the jackrabbit version. we saw issues with recent jackrabbits, but if i remember correctly that was something about CSRF token missing.

did you maybe run the tests once with a different definition of those types on your machine? once the types are stored, they cant be changed anymore. types are not per workspace but globally for the whole jackrabbit server. you could try running the jackrabbit.jar in a diffferent location (e.g. /tmp) to see if its that. also check if the names are already used somewhere else in the tests.